packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.base.tsx

```diff
@@ -108,25 +116,29 @@ export class GroupedListBase extends BaseComponent<IGroupedListProps, IGroupedLi
   }
 
   public toggleCollapseAll(allCollapsed: boolean): void {
-    const { groups } = this.state;
+    const { groups = [] } = this.state;
     const { groupProps } = this.props;
     const onToggleCollapseAll = groupProps && groupProps.onToggleCollapseAll;
 
-    if (groups) {
+    if (groups.length > 0) {
       if (onToggleCollapseAll) {
         onToggleCollapseAll(allCollapsed);
       }
 
-      for (let groupIndex = 0; groupIndex < groups.length; groupIndex++) {
-        groups[groupIndex].isCollapsed = allCollapsed;
-      }
+      this._setCollapseAllGroups(groups, allCollapsed);
 
       this._updateIsSomeGroupExpanded();
 
       this.forceUpdate();
     }
   }
 
+  private _setCollapseAllGroups(groups: IGroup[], isCollapsed: boolean): void {
+    for (let groupIndex = 0; groupIndex < groups.length; groupIndex++) {
+      groups[groupIndex].isCollapsed = isCollapsed;
```

Comment:
Wait... What about the case when we have some groups collapsed and other ones not. With this change, you disregard the ones that dev intended to have them collapsed. Am I missing something?

Reply:
I think when `isAllGroupsCollapsed` is `false` or `undefined` it does not mean all groups are necessarily expanded.

Reply:
It should change the `group.isCollapsed` for all groups on the first render only if it's `true`. Correct me if I am wrong on how I think about this prop.

Reply:
This function is not properly named, addressed in https://github.com/OfficeDev/office-ui-fabric-react/pull/8527/commits/bb7a1f16b0e586cdcc1c5e9256ea6daf6df6e8d2.

Look at the call sites for `_setGroupsCollapsedState`, it is invoked to normalize group collapsed state when "Toggle All" user interaction occurs. Now also called on initialization given the optional global toggle flag that this bug is reporting.

Reply:
> It should change the `group.isCollapsed` for all groups on the first render only if it's `true`. Correct me if I am wrong on how I think about this prop.

Isn't that guarded against by this conditional?

https://github.com/OfficeDev/office-ui-fabric-react/pull/8527/files#diff-cac9b618a0375e9796f3ee8323d63c99R78

Reply:
So I am thinking of this scenario:
Dev is creating groups with the intend to have one collapsed and the other one not on initial render.
```js
groups = {
  group1: {
    isCollapsed: true
  }
  group2: {
    isCollapsed: false
  }
}
```

If he ignores passing the `isAllGroupsCollapsed`, now the `_setGroupsCollapsedState` is going to collapse them all. Isn't this breaking expected behavior?

Reply:
NVM... thanks for pointing out the check. Missed it 😄